core/src/main/kotlin/api/RunsRoute.kt

```diff
@@ -252,6 +256,52 @@ fun Route.runs() = route("runs") {
                 }
             }
         }
+
+        route("statistics") {
+            get(getOrtRunStatistics) {
+                call.forRun(ortRunRepository) { ortRun ->
+                    requirePermission(RepositoryPermission.READ_ORT_RUNS.roleName(ortRun.repositoryId))
+
+                    val jobs = repositoryService.getJobs(ortRun.repositoryId, ortRun.index)
+
+                    val finishedStateSet: Set<JobStatus> = setOf(JobStatus.FINISHED, JobStatus.FINISHED_WITH_ISSUES)
+
+                    val analyzerJobInFinalState = jobs?.analyzer?.status in finishedStateSet.plus(JobStatus.FAILED)
+                    val analyzerJobInFinishedState = jobs?.analyzer?.status in finishedStateSet
+                    val advisorJobInFinishedState = jobs?.advisor?.status in finishedStateSet
+                    val evaluatorJobInFinishedState = jobs?.evaluator?.status in finishedStateSet
```

Comment:
If you add `== true` to the lines above you can get rid of the `takeIf`s below.

Reply:
I had some trouble with this one, these are already booleans, so adding the `== true` seems to be redundant. I tried to take the `takeIf`s away, but then it just does the counts either way, although maybe I did it somehow wrong, as I'm still quite new to Kotlin :smile:

Reply:
`if (analyzerJobInFinalState) issueService.countForOrtRunId(ortRun.id) else null` would look a lot more readable to me though, but I guess I mostly went for the takeIf/let approach, as I feel I've seen that mostly instead of if/else clauses.

Reply:
I read the code wrong, there are situations where an expression can return `null` or a `boolean` and then explicit comparison with `true` or `false` can make the code simpler, but this is not one of them.

I actually do prefer if/else over `takeIf` in most situations because I agree that it is a lot more readable.

Reply:
Ok yeah, I changed them now.